pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -290,7 +290,7 @@ private SuccessResponse uploadSegmentInternal(FormDataMultiPart multiPart, Strin
 
     try {
       FileUploadPathProvider provider = new FileUploadPathProvider(_controllerConf);
-      String tempSegmentName = "tmp-" + System.nanoTime();
+      String tempSegmentName = "tmp-" + System.currentTimeMillis();
```

Comment:
I guess our assumption (which seems fair to me) is that we won't have collision at millisecond level?

Reply:
To further ensure that, I will append nano time after current time